src/pages/javascript/jquery-widgets/quick-search.md

```diff
@@ -165,3 +165,21 @@ The result is an input with autocomplete results, where the results will be retu
 
 [`<Magento_Search_module_dir>/view/frontend/web/js/form-mini.js`]: https://github.com/magento/magento2/blob/2.4/app/code/Magento/Search/view/frontend/web/js/form-mini.js
 [Initialize JavaScript]: ../init.md
+
+## How QuickSearch Utilizes OpenSearch
+
+When a user performs a search query using QuickSearch, the input is processed by OpenSearch using its default Standard Built-in analyzers. These analyzers follow a specific set of rules:
+
+- Parses strings into tokens at word boundaries
+- Removes most punctuation
+- Converts tokens to lowercase
+
+## Hyphens in Product Names
+
+This means that for a product with the name "abcd-efg" searching with "abc" will return "abcd" as a suggestion. However, searching with "abcd" will give an empty suggestion, as the word boundary is broken with the upcoming hyphen ("-"). Additionally, this behavior affects candidate generators, as described in the OpenSearch documentation on [candidate generators](https://opensearch.org/docs/latest/search-plugins/searching-data/did-you-mean/#candidate-generators)
```

Comment:
```suggestion
If a product is named “abcd-efg”, a search for “abc” will return “abcd” as a suggestion. However, a search for “abcd” will return no suggestions because the word boundary is broken by the subsequent hyphen (`-`). This behavior also impacts OpenSearch [candidate generators](https://opensearch.org/docs/latest/search-plugins/searching-data/did-you-mean/#candidate-generators).
```

Reply:
Done